jest-common/src/main/java/io/searchbox/snapshot/AbstractSnapshotAction.java

```diff
@@ -0,0 +1,51 @@
+package io.searchbox.snapshot;
+
+import io.searchbox.action.GenericResultAbstractAction;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * @author happyprg(hongsgo@gmail.com)
+ */
+public abstract class AbstractSnapshotAction extends GenericResultAbstractAction {
+	protected Set<String> repositoryNames = new LinkedHashSet<String>();
+	protected String repositoryName;
+
+	protected AbstractSnapshotAction(Builder builder) {
+		super(builder);
+		repositoryName = builder.repositoryName;
+	}
+
+	@Override
+	protected String buildURI() {
+		StringBuilder sb = new StringBuilder("_snapshot");
+		if (StringUtils.isNotBlank(repositoryName)) {
+			sb.append("/" + repositoryName);
```

Comment:
You'll be removing this line when you modify the builder to have only the set but for feature reference: just chain the append calls instead (i.e.: `sb.append("/").append(rep)`).